### PR TITLE
[FIX] hr_payroll: salary slip refund without recompute of lines

### PR DESCRIPTION
**Pull request by jpp-odoo**
Before this commit, when a refund of a salary slip was done, the 'Salary
Computation' lines where recompute from the 'Worked Days & Inputs'
lines. This generates falsely data if the structure or the contract
changes between the salary slip and the refund.

Now, when a refund is done, all the lines are copied from the salary
slip to be coherent with the salary slip.

opw-2081454

### CODE REVIEW

File: addons/hr_payroll/models/hr_payslip.py [hunk 1]
```diff
@@ -100,8 +101,36 @@ def action_payslip_cancel(self):
     def refund_sheet(self):
         for payslip in self:
             copied_payslip = payslip.copy({'credit_note': True, 'name': _('Refund: ') + payslip.name})
-            copied_payslip.compute_sheet()
-            copied_payslip.action_payslip_done()
+            number = payslip.number or self.env['ir.sequence'].next_by_code('salary.slip')
+            lines = []
+            for l in payslip.line_ids:
```
Comment (simongoffin) [inline addons/hr_payroll/models/hr_payslip.py]: payslip.line_ids.copy({'slip_id': copied_payslip.id}) perhaps shorter

**Comment by simongoffin** [comment]
robodoo delegate+

**Comment by jpp-odoo** [comment]
robodoo r+

**Comment by MohsenWaleed** [comment]
Hi @jpp-odoo 
I'm facing this issue in Odoo11, when I do a refund, the inputs lines not included in the refund because of the copy method not copied the inputs lines to the refund.  As I can see you already fixed this in odoo 12. 

Is this fix will be applied to Odoo11?

**Pull request closed without merging**